homeassistant/components/notify/__init__.py

```diff
@@ -46,12 +49,13 @@
 NOTIFY_SERVICE_SCHEMA = vol.Schema({
     vol.Required(ATTR_MESSAGE): cv.template,
     vol.Optional(ATTR_TITLE): cv.template,
+    vol.Optional(ATTR_MEDIA): cv.isfile,
```

Comment:
Other notify platforms are using `ATTR_DATA` for it. That way if a platform requires extra metadata, we don't need to add extra fields. So please add it to Twitter via `ATTR_DATA`.